charts/consul/templates/crd-httprouteretryfilters.yaml

```diff
@@ -0,0 +1,122 @@
+{{- if .Values.connectInject.enabled }}
+---
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  annotations:
+    controller-gen.kubebuilder.io/version: v0.9.2
+  creationTimestamp: null
+  name: httprouteretryfilters.consul.hashicorp.com
+  labels:
+    app: {{ template "consul.name" . }}
+    chart: {{ template "consul.chart" . }}
+    heritage: {{ .Release.Service }}
+    release: {{ .Release.Name }}
+    component: crd
+spec:
+  group: consul.hashicorp.com
+  names:
+    kind: HTTPRouteRetryFilter
+    listKind: HTTPRouteRetryFilterList
+    plural: httprouteretryfilters
+    singular: httprouteretryfilter
+  scope: Namespaced
+  versions:
+  - additionalPrinterColumns:
+    - description: The sync status of the resource with Consul
+      jsonPath: .status.conditions[?(@.type=="Synced")].status
+      name: Synced
+      type: string
+    - description: The last successful synced time of the resource with Consul
+      jsonPath: .status.lastSyncedTime
+      name: Last Synced
+      type: date
+    - description: The age of the resource
+      jsonPath: .metadata.creationTimestamp
+      name: Age
+      type: date
+    name: v1alpha1
+    schema:
+      openAPIV3Schema:
+        description: HTTPRouteRetryFilter is the Schema for the httprouteretryfilters
+          API
+        properties:
+          apiVersion:
+            description: 'APIVersion defines the versioned schema of this representation
+              of an object. Servers should convert recognized schemas to the latest
+              internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources'
+            type: string
+          kind:
+            description: 'Kind is a string value representing the REST resource this
+              object represents. Servers may infer this from the endpoint the client
+              submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds'
+            type: string
+          metadata:
+            type: object
+          spec:
+            description: HTTPRouteRetryFilterSpec defines the desired state of HTTPRouteRetryFilter
+            properties:
+              numRetries:
+                format: int32
+                type: integer
+              retryOn:
+                items:
+                  type: string
+                type: array
+              retryOnConnectFailure:
+                type: boolean
+              retryOnStatusCodes:
+                items:
+                  format: int32
+                  type: integer
+                type: array
+            required:
+            - numRetries
+            - retryOn
+            - retryOnConnectFailure
+            - retryOnStatusCodes
```

Comment:
I don't think it makes sense to require all of these be set since you could reasonably set up connect failure retry without setting up status code retry